modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java

```diff
@@ -190,19 +202,50 @@ protected synchronized void open() {
 
         // Open the current log file
         try {
-            String pathname;
+            Date today = new Date(System.currentTimeMillis());
+            String pathname = dir.getAbsolutePath() + File.separator + AccessConstants.getPrefix() +
+                    AccessConstants.getSuffix();
+            String oldPathname;
             // If no rotate - no need for dateStamp in fileName
             if (isRotatable) {
-                pathname = dir.getAbsolutePath() + File.separator + AccessConstants.getPrefix() +
-                           dateStamp + AccessConstants.getSuffix();
-            } else {
-                pathname = dir.getAbsolutePath() + File.separator + AccessConstants.getPrefix() +
-                           AccessConstants.getSuffix();
+                File existing = new File(pathname);
+                // renaming existing file to a file with date
+                if (existing != null && existing.exists()) {
```

Comment:
existing will never be null, right?

Reply:
can be, when the server start for the first time

Reply:
+1 filed